File: 151001/Shcherba/lab2/src/main/java/by/bsuir/entities/Tweet.java
package by.bsuir.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "tbl_tweet", schema = "distcomp")
public class Tweet {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;
    @Column(name = "title")
    private String title;
    @Column(name = "content")
    private String content;
    @Column(name = "created")
    private Timestamp created;
    @Column(name = "modified")
    private Timestamp modified;
    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

}

File: 151003/Klimovich/Publisher/Models/Tweet.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace Publisher.Models
{
    [Table("tbl_tweet")]
    public class Tweet
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Required]
        public int id { get; set; }


        [MaxLength(64)]
        public string title { get; set; }

        [MaxLength(2048)]
        public string content { get; set; }


        public DateTime created { get; set; }


        public DateTime modified { get; set; }

        [Required]
        [ForeignKey("User")]
        public int userId { get; set; }
        public User User { get; set; }
        public List<Comment>? Comments { get; set; } = new();
        public List<Sticker> Stickers { get; set; } = new();
        public List<TweetSticker> TweetStickers { get; set; } = new();

    }
}
